zenergy: Build fails with implicit declaration of function rdmsrl_safe/rdmsrl_safe_on_cpu

### Nixpkgs version

- Unstable (25.11)

### Describe the bug

Using flake with unstable packages.

__Commit:__ dfb2f12e899db4876308eba6d93455ab7da304cd
__narHash:__ sha256-1wxxznpW2CKvI9VdniaUnTT2Os6rdRJcRUf65ZK9OtE=

---

It is not possible to rebuild the system with the `zenergy` module.

I found another Issue ([Broken with 6.16.0-rc1 #15](https://github.com/BoukeHaarsma23/zenergy/issues/15#issuecomment-3094632391)) that found the problem, but it hasn't been officially patched in a while.

__EDIT:__ And here is a linked [PR with a potential solution](https://github.com/BoukeHaarsma23/zenergy/pull/18#issuecomment-3141015950).
<details>
<summary>Temporary workaround to build</summary>

```SH
extraModulePackages = with config.boot.kernelPackages; [
  (config.boot.kernelPackages.zenergy.overrideAttrs (prev: {
    # https://github.com/BoukeHaarsma23/zenergy/issues/15#issuecomment-3094632391
    patches = (prev.patches or [ ]) ++ [
      (pkgs.fetchpatch {
        # https://github.com/BoukeHaarsma23/zenergy/pull/18#issuecomment-3141015950
        url = "https://patch-diff.githubusercontent.com/raw/BoukeHaarsma23/zenergy/pull/18.patch";
        hash = "sha256-Xe6CaiU9Vcxxgzu4VGrriq5sAQfymFbsw4kEf3Gj2C0=";
      })
    ];
  }))
];
```
</details>

__EDIT 2:__ Even with this temporary solution, Mangohud can't access the CPU temperature. However, since I can see a temperature in `btop`, I'm not sure if this is a bug in Mangohud or if some output from `zenergy` is missing.
```SH
[2025-08-30 14:55:38.972] [MANGOHUD] [error] [cpu.cpp:552] Could not find cpu temp sensor location
```

### Steps to reproduce

Rebuilding system flake with `nixos-rebuild`.

### Expected behaviour

Successful system build with zenergy

### Screenshots

_No response_

### Relevant log output

```console
$> nix log /nix/store/gmh277zhjwj093qsay0rqbqama9q1i92-zenergy-0-unstable-2025-04-15.drv

Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/pg4mi4vn4wpas8xvl8gvik87rz8894qr-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/4bacfs7zrg714ffffbjp57nsvcz6zfkq-bash-5.3p3/bin/bash CC=/nix/store/qdknxw57cwy1jkrhq7fzmiis73j42jv6-gcc-14.3.0/bin/gcc LD=/nix/store/xrwdb41dqi2ia6lr2s61w5bzfg2m71pi-binutils-2.44/bin/ld AR=/nix/store/bcw9f6r9v2fm3kv7d15fcrya0mf34xds-gcc-wrapper-14.3.0/b>
export CONFIG_SENSOR_zenergy=m; \
make -C /nix/store/vrvgik2kfm2bmb2wjvrpqka933w5pg8w-linux-6.16.3-dev/lib/modules/6.16.3/build M=$PWD modules
make[1]: Entering directory '/nix/store/vrvgik2kfm2bmb2wjvrpqka933w5pg8w-linux-6.16.3-dev/lib/modules/6.16.3/build'
make[2]: Entering directory '/build/source'
  CC [M]  zenergy.o
zenergy.c: In function 'get_energy_units':
zenergy.c:83:9: error: implicit declaration of function 'rdmsrl_safe'; did you mean 'rdmsrq_safe'? [-Wimplicit-function-declaration]
   83 |         rdmsrl_safe(ENERGY_PWR_UNIT_MSR, &rapl_units);
      |         ^~~~~~~~~~~
      |         rdmsrq_safe
zenergy.c: In function '__accumulate_delta':
zenergy.c:92:9: error: implicit declaration of function 'rdmsrl_safe_on_cpu'; did you mean 'rdmsrq_safe_on_cpu'? [-Wimplicit-function-declaration]
   92 |         rdmsrl_safe_on_cpu(cpu, reg, &input);
      |         ^~~~~~~~~~~~~~~~~~
      |         rdmsrq_safe_on_cpu
make[4]: *** [/nix/store/vrvgik2kfm2bmb2wjvrpqka933w5pg8w-linux-6.16.3-dev/lib/modules/6.16.3/source/scripts/Makefile.build:287: zenergy.o] Error 1
make[3]: *** [/nix/store/vrvgik2kfm2bmb2wjvrpqka933w5pg8w-linux-6.16.3-dev/lib/modules/6.16.3/source/Makefile:2003: .] Error 2
make[2]: *** [/nix/store/vrvgik2kfm2bmb2wjvrpqka933w5pg8w-linux-6.16.3-dev/lib/modules/6.16.3/source/Makefile:248: __sub-make] Error 2
make[2]: Leaving directory '/build/source'
make[1]: *** [/nix/store/vrvgik2kfm2bmb2wjvrpqka933w5pg8w-linux-6.16.3-dev/lib/modules/6.16.3/source/Makefile:248: __sub-make] Error 2
make[1]: Leaving directory '/nix/store/vrvgik2kfm2bmb2wjvrpqka933w5pg8w-linux-6.16.3-dev/lib/modules/6.16.3/build'
make: *** [Makefile:17: default] Error 2
```

### Additional context

```nix
boot = {
  blacklistedKernelModules = [
    "k10temp"
  ];
  kernelPackages = pkgs.linuxPackages_latest;
  initrd = {
    availableKernelModules = [
      "nvme"
      "xhci_pci"
      "ahci"
      "usb_storage"
      "usbhid"
      "sd_mod"
      "sr_mod"
    ];
    kernelModules = [
      "ntsync"
      "nvidia"
      "zenergy"
    ];
    systemd.enable = true;
  };
  kernelModules = [ "kvm-amd" ];
  extraModulePackages = with config.boot.kernelPackages; [
    nvidia_x11
    zenergy
  ];
}
```

### System metadata

```SH
$> nix-shell -p nix-info --run "nix-info -m"

 - system: `"x86_64-linux"`
 - host os: `Linux 6.16.3, NixOS, 25.11 (Xantusia), 25.11.20250828.dfb2f12`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.4`
 - nixpkgs: `/nix/store/4hm8lf740i8qvyg5pzdqfm0rpshwb7vn-source`
```

### Notify maintainers

@wizardlink 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/zenergy/default.nix
+++ b/pkgs/os-specific/linux/zenergy/default.nix
@@ -12,13 +12,13 @@ let
 in
 stdenv.mkDerivation {
   pname = "zenergy";
-  version = "0-unstable-2025-04-15";
+  version = "0-unstable-2025-08-31";
 
   src = fetchFromGitHub {
     owner = "BoukeHaarsma23";
     repo = "zenergy";
-    rev = "f77293fc4aa8c2f5645b2d05d8f0d476220cba9a";
-    hash = "sha256-T9ualNYna2Ip19dqz1mOcFWX5oKWIhf9SGMaXovS8QE=";
+    rev = "58f2fda7184fbde95033f492f7c54990552ef86f";
+    hash = "sha256-nSkq4JuZqhuH+JGH/vr9bw/suo/2rmdbKcvYPIil9qw=";
   };
 
   nativeBuildInputs = [ kmod ] ++ kernel.moduleBuildDependencies;

